engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetViewportToSizeOf.java

```diff
@@ -65,12 +69,20 @@ public RenderPipelineTask generateTask() {
 
     @Override
     public int hashCode() {
-        return Objects.hashCode(fboName);
+        // Generates a unique 32 bit signed integer from two integers.
+        // This will return unique values for the following reasonable ranges:
+        // width < (1 << 16) (65536) and height < (1 << 15) (32768)
+        return getWidth() << 15 | getHeight();
```

Comment:
The ranges _are_ reasonable, but why impose arbitrary restrictions when http://docs.oracle.com/javase/7/docs/api/java/util/Objects.html#hash(java.lang.Object...) is a thing? 😛